include/ccf/pal/snp_ioctl6.h

```diff
@@ -140,4 +197,46 @@ namespace ccf::pal::snp::ioctl6
       return {quote_bytes, quote_bytes + resp_wrapper.resp.report_size};
     }
   };
-}
+
+  class DerivedKey
+  {
+    DerivedKeyRespWrapper resp_wrapper = {};
+
+  public:
+    DerivedKey()
+    {
+      int fd = open(DEVICE, O_RDWR | O_CLOEXEC);
+      if (fd < 0)
+      {
+        throw std::logic_error(fmt::format("Failed to open \"{}\"", DEVICE));
+      }
+
+      DerivedKeyReq req = {};
+      GuestRequestDerivedKey payload = {
+        .req_data = &req, .resp_wrapper = &resp_wrapper, .exit_info = {0}};
+      int rc = ioctl(fd, SEV_SNP_GUEST_MSG_DERIVED_KEY, &payload);
+      if (rc < 0)
+      {
+        CCF_APP_FAIL("IOCTL call failed: {}", strerror(errno));
```

Comment:
```suggestion
        LOG_FAIL_FMT("IOCTL call failed: {}", strerror(errno));
```

CCF_APP is for app code